modules/videoio/src/cap_dshow.cpp

```diff
@@ -2946,6 +2946,22 @@ int videoInput::start(int deviceID, videoDevice *VD){
     DebugPrintOut("SETUP: Device is setup and ready to capture.\n\n");
     VD->readyToCapture = true;
 
+    // check for optional saving the direct show graph to a file
+    const char* graph_filename = getenv("OPENCV_DSHOW_SAVEGRAPH_FILENAME");
+    if (graph_filename) {
+        WCHAR* wfilename = new WCHAR[strlen(graph_filename) + 1];
```

Comment:
> new

Manual memory management should be avoided in OpenCV code.
Use `std::vector<WCHAR>` or `cv::AutoBuffer<WCHAR>` instead.

Reply:
Done